### ISSUE TITLE
[Switch] Enhance the component's colors for a11y and usability

### ISSUE THREAD

**Issue by mjuniper** [Esri/calcite-design-system#10483]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Summary

We got a number of accessibility violations on calcite-switch when LevelAccess audited Hub. See:
- https://devtopia.esri.com/dc/hub/issues/11268
- https://devtopia.esri.com/dc/hub/issues/11241
- https://devtopia.esri.com/dc/hub/issues/11273
- https://devtopia.esri.com/dc/hub/issues/11288
- https://devtopia.esri.com/dc/hub/issues/11300
 
I think these boil down to: when the switch is off the track border (`--calcite-color-border-2` / `#d4d4d4`) does not sufficiently contrast with the track background (`--calcite-color-foreground-2` / `#f3f3f3`). It would not sufficiently contrast with even a white (#ffffff) surrounding background either but I think it is sufficient to focus on making its own border and background contrast sufficiently.

### Actual Behavior

calcite switch has insufficient contrast between the track border and track background when in the off position.

![Blank diagram (5)](https://github.com/user-attachments/assets/109a9956-30af-461f-8da4-c3fac4a15a28)

### Expected Behavior

calcite-switch should have sufficient contrast between the track border and track background when in the off position. I think something like border: #949494 (--calcite-color-border-input) and background #ffffff (--calcite-color-foreground-1) would do it.

### Calcite solution

- [x] Increase contrast and visual presence of idle state
- [x] Simplify track and handle to one color and remove strokes
- [x] Simplify hover & focus state: remove hover and focus ring on the handle

🔗[ Link to Figma specs](https://www.figma.com/design/0sgmjyobwIcUm7QoaBB1pJ/%5BSwitch%5D-Enhance-the-component's-colors-for-a11y-and-usability?node-id=3501-16909&m=dev&focus-id=6501-537 )   🔗[Link to prototype](https://www.figma.com/proto/0sgmjyobwIcUm7QoaBB1pJ/%5BSwitch%5D-Enhance-the-component's-colors-for-a11y-and-usability?page-id=3501%3A16909&node-id=6512-319&viewport=7125%2C-3183%2C0.89&t=TkQrBGomFJmIqnwe-1&scaling=min-zoom&content-scaling=fixed)

<img width="1058" alt="Image" src="https://github.com/user-attachments/assets/52862dcb-a11d-42a3-898c-b3ad45807432" />

### Reproduction Sample

https://codepen.io/mjuniper/pen/rNXemPZ?editors=100

### Reproduction Steps

1. Visit https://codepen.io/mjuniper/pen/rNXemPZ?editors=100
2. Observe that the contrast between the track border (#d4d4d4) and the track background (#f3f3f3) is only 1.3
3. Also observe that the contrast between the track border and the white surrounding background is only 1.48.

### Reproduction Version

2.13.0

### Priority impact

impact - p1 - need for current milestone

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-angular
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Hub

**monday.com sync:** #8135677069

**Comment by geospatialem** [comment]
Adding design for expertise on any modifications to the switch component's contrast when in the "off" position.

**Comment by DitwanP** [comment]
🍡 Verified locally on `dev`
